src/KeywordExtractor/KeywordExtractor.php

```diff
@@ -64,6 +64,54 @@ public function run(string $string, string $sortBy = '', string $sortDir = Sorte
         return $this->keywords;
     }
 
+    public function getKeywords(): array
+    {
+        return $this->retrieveKeywords();
+    }
+
+    /**
+     * Retrieve flattened and clean keywords. The process includes the following steps:
+     * - flatten the keywords
+     * - trim punctuation marks
+     * - remove duplicates
+     *
+     * @return array
+     */
+    private function retrieveKeywords(): array
+    {
+        if (empty($this->keywords)) {
+            return [];
+        }
+
+        $keywords = array_reduce($this->keywords, function ($flattened, $structure) {
+            $shortest = $this->findShortest($structure['occurrences']);
+            $trimmed = preg_replace('/^[^\p{L}0-9]+|[^\p{L}0-9]+\z/u', '', $shortest);
```

Comment:
For consistency, I think we can centralise the trimming logic somewhere since we have a similar implementation in `PunctuationFilter`. It's fine for now.

Reply:
This trimming logic is for cases when there is only one occurrence of a word and this word is near a punctuation mark.
In other words, imagine the input is `This is a simple sequence of sentences.`. The result for this input is going to be:
```
array(3) {
  ["simpl"]=>
  array(2) {
    ["frequency"]=>
    int(1)
    ["occurrences"]=>
    array(1) {
      [0]=>
      array(2) {
        ["ngram"]=>
        string(6) "simple"
        ["indexes"]=>
        array(1) {
          [0]=>
          int(3)
        }
      }
    }
  }
  ["sequenc"]=>
  array(2) {
    ["frequency"]=>
    int(1)
    ["occurrences"]=>
    array(1) {
      [0]=>
      array(2) {
        ["ngram"]=>
        string(8) "sequence"
        ["indexes"]=>
        array(1) {
          [0]=>
          int(4)
        }
      }
    }
  }
  ["sentenc"]=>
  array(2) {
    ["frequency"]=>
    int(1)
    ["occurrences"]=>
    array(1) {
      [0]=>
      array(2) {
        ["ngram"]=>
        string(10) "sentences." // this includes a punctuation mark
        ["indexes"]=>
        array(1) {
          [0]=>
          int(6)
        }
      }
    }
  }
}```